app/repositories/ZonedDateTimeFormat.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import play.api.libs.json._
+
+import java.time.{Instant, ZoneId, ZoneOffset, ZonedDateTime}
+import scala.util.Try
+
+object ZonedDateTimeFormat {
+
+  lazy val zonedDateTimeReads: Reads[ZonedDateTime] =
+    fromDateObj.orElse(fromNumberLongObj).orElse(fromDateTimeStringWithZone)
+
+  val zonedDateTimeWrites: Writes[ZonedDateTime] =
+    instantWrites.contramap(_.withZoneSameInstant(ZoneId.of("UTC")).toInstant)
+
+  lazy val zonedDateTimeFormat: Format[ZonedDateTime] = Format(zonedDateTimeReads, zonedDateTimeWrites)
+
+  private val fromDateObj: Reads[ZonedDateTime] =
+    Reads.at[Long](__ \ "$date").map(Instant.ofEpochMilli(_).atZone(ZoneOffset.UTC))
+
+  private val fromNumberLongObj: Reads[ZonedDateTime] =
+    Reads.at[String](__ \ "$date" \ "$numberLong").map(s => Instant.ofEpochMilli(s.toLong).atZone(ZoneOffset.UTC))
+
+  private val fromDateTimeStringWithZone: Reads[ZonedDateTime] =
+    (value: JsValue) =>
+      Try {
+        JsSuccess(ZonedDateTime.parse(value.as[String]))
+      }.getOrElse(JsError(s"[$value] cannot be parsed to ZonedDateTime"))
+
+  lazy val instantWrites: Writes[Instant] = Writes.at[Long](__ \ "$date").contramap(_.toEpochMilli)
+}
```

Comment:
This all makes sense but I do think that we should create a migration job to convert all the dateTime values in our collections to the new ZonedDateTime format. 

I think consistency of the format of the dateTime fields across all documents in a particular collection is important. If we ever needed to perform a query that used one of the dateTime fields as a predicate and they were inconsistent it would be a nightmare.

Reply:
I'm ok with that but let's do it with another ticket. For the time being, but honestly I dont even foresee any future problems in this regard, we are safe since what is resulting from our `Writes` is what is currently saved in our collections.

Reply:
But all the documents that have already been saved to the collections in QA. ET & Prod will still retain the old formats. If we were to deploy this branch before the migration job we would get inconsistent document formats in these environment's collections :frowning_face:

Reply:
But the output format (what we have in the DB) is not changed. The Joda DateTime is not represented in the DB. This was just the conversion type when **reading** from the DB. To give you an example. This is what we have locally in the DB after my changes
collection(exportsMigrationChangeLog) => "timestamp" : ISODate("2021-06-07T12:07:15.325Z")

and this is what we have in QA
collection(exportsMigrationChangeLog) => "timestamp" : ISODate("2021-01-04T15:53:30.316Z")

and so forth for "notifications" and "answers"